src/collections/map.rs

```diff
@@ -128,6 +129,44 @@ impl<K, V, const PREFIX_LEN: usize> TreeMap<K, V, PREFIX_LEN> {
         }
     }
 
+    /// Constructs a [`TreeMap`] from a raw node pointer.
+    ///
+    /// # Safety
+    ///
+    ///  - The raw pointer must have been previously returned by a call to
+    ///    [`TreeMap::into_raw_with_allocator`] or [`TreeMap::into_raw`].
+    ///     - The allocator of the previous tree must have been the "default"
+    ///       allocator named `Global`.
+    ///  - The given `root` pointer must be unique and there are no other
+    ///    pointers into the tree.
+    ///  - `root` must be a pointer to a well formed tree.
+    ///
+    /// # Examples
+    ///
+    /// ```rust
+    /// use blart::TreeMap;
+    ///
+    /// let mut map = TreeMap::<Box<[u8]>, char>::new();
+    ///
+    /// map.try_insert(Box::new([1, 2, 3]), 'a').unwrap();
+    /// assert_eq!(map.len(), 1);
+    ///
+    /// let root = TreeMap::into_raw(map);
+    /// assert!(root.is_some());
+    ///
+    /// // SAFETY: The root pointer came directly from the `into_raw` result.
+    /// let _map = unsafe { TreeMap::unchecked_from_raw(root) };
+    /// ```
+    pub unsafe fn unchecked_from_raw(root: Option<OpaqueNodePtr<K, V, PREFIX_LEN>>) -> Self
```

Comment:
```suggestion
    /// let _map = unsafe { TreeMap::from_raw_unchecked(root) };
    /// ```
    pub unsafe fn from_raw_unchecked(root: Option<OpaqueNodePtr<K, V, PREFIX_LEN>>) -> Self
```